### PR TITLE
*: add where for show table region syntax

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
* Add `where` for `show table region` syntax

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -709,8 +709,10 @@ AAAAAAAAAAAA5gm5Mg==
 
 		// for show table regions.
 		{"show table t1 regions", true, "SHOW TABLE `t1` REGIONS"},
+		{"show table t1 regions where a=1", true, "SHOW TABLE `t1` REGIONS WHERE `a`=1"},
 		{"show table t1", false, ""},
 		{"show table t1 index idx1 regions", true, "SHOW TABLE `t1` INDEX `idx1` REGIONS"},
+		{"show table t1 index idx1 regions where a=2", true, "SHOW TABLE `t1` INDEX `idx1` REGIONS WHERE `a`=2"},
```
Comment (kennytm) [inline parser_test.go]: Also add two test cases for `show table t1 regions like 'something%'`?

**Comment by winkyao** [comment]
What is this PR used for？

**Comment by crazycs520** [comment]
@winkyao This  use for support SQL like `show table t regions where scattering=1;`

File: parser_test.go (hunk 1 above)
Comment (crazycs520) [reply parser_test.go]: done.

Comment (kennytm) [reply parser_test.go]: @crazycs520 I mean literal ``SHOW TABLE `t1` INDEX `idx1` REGIONS LIKE 'some%'``, not ``WHERE `a` LIKE 'some%'``. Don't use `ShowLikeOrWhereOpt` if you can't support that...

Comment (crazycs520) [reply parser_test.go]: Great, thanks. PTAL

File: parser.y [hunk 2]
```diff
@@ -6886,20 +6886,33 @@ ShowStmt:
...
 			Tp:	ast.ShowRegions,
 			Table:	$3.(*ast.TableName),
 		}
+		if $5 != nil {
+			stmt.Where = $5.(ast.ExprNode)
+		}
+		$$ = stmt
 	}
-|	"SHOW" "TABLE" TableName "INDEX" Identifier "REGIONS"
+|	"SHOW" "TABLE" TableName "INDEX" Identifier "REGIONS" ShowLikeOrWhereOpt
```
Comment (kennytm) [inline parser.y]: Should this still be `ShowLikeOrWhereOpt`?

Comment (crazycs520) [reply parser.y]: Done. Thanks.

File: parser.y [hunk 3]
```diff
@@ -7612,20 +7612,33 @@ ShowStmt:
...
 	{
-		$$ = &ast.ShowStmt{
+		stmt := &ast.ShowStmt{
 			Tp:	ast.ShowRegions,
 			Table:	$3.(*ast.TableName),
 			IndexName: model.NewCIStr($5),
 		}
+		if $7 != nil {
+			if x, ok := $7.(*ast.PatternLikeExpr); ok && x.Expr == nil {
+				stmt.Pattern = x
```
Comment (kennytm) [inline parser.y]: Remove this branch.

Comment (crazycs520) [reply parser.y]: Done. Thanks.

**Comment by crazycs520** [comment]
@kennytm PTAL

**Pull request merged**